src/jit/codegenxarch.cpp

```diff
@@ -7366,6 +7366,66 @@ void CodeGen::genSSE2BitwiseOp(GenTreePtr treeNode)
     inst_RV_RV(ins, targetReg, operandReg, targetType);
 }
 
+//-----------------------------------------------------------------------------------------
+// genSSE41RoundOp - generate SSE41 code for the given tree as a round operation
+//
+// Arguments:
+//    treeNode  - tree node
+//
+// Return value:
+//    None
+//
+// Assumptions:
+//     i) SSE4.1 is supported by the underlying hardware
+//    ii) tree oper is a GT_INTRINSIC
+//   iii) tree type is floating point type
+//    iv) tree oper is CORINFO_INTRINSIC_Round, _Ceiling, or _Floor
+//     v) caller of this routine needs to call genProduceReg()
+void CodeGen::genSSE41RoundOp(GenTree* treeNode)
+{
+    // i) SSE4.1 is supported by the underlying hardware
+    assert(compiler->compSupports(InstructionSet_SSE41));
+
+    // ii) tree oper is a GT_INTRINSIC
+    assert(treeNode->OperGet() == GT_INTRINSIC);
+
+    GenTree*    srcNode  = treeNode->gtGetOp1();
+
+    instruction ins      = ins_FloatRound(treeNode->TypeGet());
+    emitAttr    size     = emitTypeSize(treeNode);
+
+    regNumber   dstReg   = treeNode->gtRegNum;
+    regNumber   srcReg   = srcNode->gtRegNum;
+
+    // iii) tree type is floating point type
+    assert(varTypeIsFloating(srcNode));
+    assert(srcNode->TypeGet() == treeNode->TypeGet());
+
+    genConsumeOperands(treeNode->AsOp());
+
+    // iv) tree oper is CORINFO_INTRINSIC_Round, _Ceiling, or _Floor
+    switch (treeNode->gtIntrinsic.gtIntrinsicId)
+    {
+        case CORINFO_INTRINSIC_Round:
+            inst_RV_RV_IV(ins, size, dstReg, srcReg, 4);
```

Comment:
It may be better to sink the `inst_RV_RV_IV` call out of the switch and use the switch only to select the rounding mode value.

Reply:
I had thought about doing that as well. Will fix.

Reply:
Fixed.